atomix/cluster/src/test/java/io/atomix/raft/RaftFailOverTest.java

```diff
@@ -170,6 +170,28 @@ public void shouldReplicateSnapshotOnJoin() throws Exception {
     assertThat(snapshot.getTerm()).isEqualTo(snapshot.getTerm());
   }
 
+  @Test
+  public void shouldReplicateSnapshotWithManyFilesOnJoin() throws Exception {
+    // given
+    final var follower = raftRule.shutdownFollower();
+    raftRule.appendEntries(20);
+    final long snapshotIndex = 10L;
+    raftRule.doSnapshot(snapshotIndex, 10);
+    final var leaderSnapshot = raftRule.getSnapshotFromLeader();
+
+    // when
+    raftRule.joinCluster(follower);
+
+    // then
+    assertThat(raftRule.allNodesHaveSnapshotWithIndex(snapshotIndex)).isTrue();
+    final var snapshot = raftRule.getSnapshotOnNode(follower);
+
+    assertThat(snapshot.getIndex()).isEqualTo(leaderSnapshot.getIndex()).isEqualTo(snapshotIndex);
+    assertThat(snapshot.getTerm()).isEqualTo(snapshot.getTerm());
+    assertThat(snapshot.getPath().toFile().list())
```

Comment:
Could be flaky, since ordering is not guaranteed right?

Reply:
First I thought hm this should also worked before right? So what is the matter of this test, the I tried it without your fix and it toooook soo long, wow didn't expected that :D 

Thanks for fixing that, awesome.

Reply:
Actually, this test was added to test parallel/pipelined sending of snapshot chunks. But this PR doesnot have it. I didn't notice performance difference :smile: Should I keep/remove this test?

Reply:
Lets keep it. It fails, because of timeouts, if I we create always the snapshot chunk reader :) I tried that :D